framework/src/media/audio/audio_manager.c

```diff
@@ -123,15 +129,37 @@ audio_manager_result_t find_input_audio_card(void)
 			|| (dir_entry->d_name[1] != 'c')
 			|| (dir_entry->d_name[2] != 'm')
 			|| (sscanf(&dir_entry->d_name[3], "C%uD%u%c", &card_id, &device_id, &type) != 3)) {
-			closedir(dir_info);
-			return AUDIO_MANAGER_INVALID_DEVICE_NAME;
+			continue;
 		}
 
 		if ((type == 'c') && (card_id < CONFIG_AUDIO_MAX_INPUT_CARD_NUM) && (device_id < MAX_AUDIO_DEVICE_NUM)) {
+#ifndef CONFIG_AUDIO_EXCLUDE_VOLUME
```

Comment:
Here, we should do manythings.
1. get volume level & max volume
2. get Supported Samplerate Type.
So I think you had better make internal API to get them easily.

Reply:
That's good.
But how about adding the API when I upload a PR for resampling?

Reply:
Okay. Let me add the internal API in the PR for resampling...